index.js

```diff
@@ -61,26 +64,46 @@ Queue.prototype.poll = function () {
       setTimeout(self.poll.bind(self), 1000 * 5)
       return
     }
-    (data.Messages || []).forEach(self._processMsg.bind(self))
-    self.poll()
+
+    var killed = false
+    var messageFinishTimeout = setTimeout(function () {
```

Comment:
@iuriaranda Thanks a lot for the pull request :+1: 

What's the reason for adding the timeout? As far as I understand it, SQS long polling will return an empty list if no messages are available after a maximum of 20 seconds. It might be perfectly fine, I just wanted to hear your thoughts :smiley: